endpoints-framework/src/main/java/com/google/api/server/spi/dispatcher/PathDispatcher.java

```diff
@@ -42,12 +42,14 @@ private PathDispatcher(Builder<ContextT> builder) {
   public boolean dispatch(String httpMethod, String path, ContextT context) throws IOException {
     Preconditions.checkNotNull(httpMethod, "httpMethod");
     Preconditions.checkNotNull(path, "path");
-    Result<DispatcherHandler<ContextT>> result =
-        trie.resolve(HttpMethod.valueOf(httpMethod.toUpperCase()), path);
-    if (result != null) {
-      context.setRawPathParameters(result.getRawParameters());
-      result.getResult().handle(context);
-      return true;
+    HttpMethod method = HttpMethod.fromString(httpMethod.toUpperCase());
```

Comment:
Why not put the `toUpperCase` inside the `fromString` method, so it doesn't matter what case the method is in?